manager/encryption/encryption.go

```diff
@@ -0,0 +1,132 @@
+package encryption
+
+import (
+	"crypto/rand"
+	"encoding/base64"
+	"fmt"
+	"io"
+	"strings"
+
+	"github.com/docker/swarmkit/api"
+	"github.com/gogo/protobuf/proto"
+	"github.com/pkg/errors"
+)
+
+// This package defines the interfaces and encryption package
+
+const humanReadablePrefix = "SWMKEY-1-"
+
+// ErrCannotDecode is the type of error returned when some data cannot be decoded as plaintext
+type ErrCannotDecode struct {
```

Comment:
This looks reasonable: https://github.com/docker/swarmkit/pull/1598/files#diff-8a1870ff8ffb551be79bf7069e478962R171.

Make sure you wrap in errors.Cause, in case we wrap the error.